### PR TITLE
feat: T2S paperdownload tools

### PR DESCRIPTION
**Pull request by ansh-info**
# For authors

## Description

Added the new unified paper download tool, with abstract base class, now we have a single unified tool.

We support the following and for testing I used the local build of the python package `aiagents4pharma`, one additional note now we actually download pdf and store it in a temp folder, in order to speed up the process for RAG pipeline

1. Arxiv - Direct API support
image
image
image
 
2. Medriv - Direct API support 
image

3. Biorxiv - Addition browser automation has been added to mimic a browser in order to download pdf via script
image

4. Pubmed - Multiple fallback needed to be added in order to support pdf download from different hosting providers, and an additional logic as been added to convert the DOI available with the paper to PMID and then download paper, and some of the papers are published in a journal with the respective DOI as oppose to the preprint DOI
image

@gurdeep330 @lilijap 
## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests you conducted to verify your changes. These may involve creating new test scripts or updating existing ones.
- [x] Added new test(s) in the ```tests``` folder
- [x] Added new function(s) to an existing test(s) (e.g.: ```tests/testX.py```)
- [ ] No new tests added (Please explain the rationale in this case)

## Checklist
- [x] My code follows the style guidelines mentioned in the Code/DevOps guides
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. MkDocs)
- [x] My changes generate no new warnings
- [x] I have added or updated tests (in the ```tests``` folder) that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?  
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

### CODE REVIEW

**Review by gurdeep330 — CHANGES_REQUESTED** [review]
Thanks for the update @ansh-info . Some minor comments.

File: aiagents4pharma/talk2scholars/configs/tools/paper_download/default.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    identifier_type: "PMID"
+    supports_batch: true
+    log_response_preview_chars: 500 # chars to log from API responses
+
+# Tool configuration
+tool:
+  name: "download_papers"
+  description: "Universal paper download tool supporting arXiv, medRxiv, bioRxiv, and PubMed"
+  supported_services: ["arxiv", "medrxiv", "biorxiv", "pubmed"]
+  default_service: "arxiv"
```
[the remark this answers is not in the document]
Comment (ansh-info) [reply aiagents4pharma/talk2scholars/configs/tools/paper_download/default.yaml]: Thank you for the review, I have updated it to pubmed

File: aiagents4pharma/talk2scholars/tools/paper_download/utils/arxiv_downloader.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+        Args:
+            metadata: XML root from arXiv API
+            identifier: arXiv ID
+            pdf_result: Tuple of (temp_file_path, filename) if PDF downloaded
+
+        Returns:
+            Standardized paper metadata dictionary
+        """
+        ns = {"atom": "http://www.w3.org/2005/Atom"}
```
[the remark this answers is not in the document]
Comment (ansh-info) [reply aiagents4pharma/talk2scholars/tools/paper_download/utils/arxiv_downloader.py]: sure
